src/fsharp/CompileOptions.fs

```diff
@@ -923,7 +912,7 @@ let advancedFlagsBoth tcConfigB =
 
         yield CompilerOption
                  ("targetprofile", tagString,
-                  OptionString (SetTargetProfile tcConfigB), None,
+                  OptionString (fun _ -> ()), None, // TODO: Add deprecation message.
```

Comment:
"--targetprofile is now deprecated and will have no effect. The F# compiler discovers the appropriate target framework profile automatically."